drivers/optical/pmw3360_firmware.h

```diff
@@ -18,6 +18,9 @@
 
 #pragma once
 
+#define PMW3360_PRODUCT_ID         0x42
```

Comment:
....where did this firmware originate? A firmware blob without a source seems...questionable.

Reply:
Well, fun times.  The firmware blob is from here: https://github.com/mrjohnk/PMW3360DM-T2QU

Which appears to be the repo for the tindie shop seller for the sensor.  So "without a source" seems to be the real answer.  Though, at least, should include said links in the file.